### PR TITLE
Fix fft_dly

### PR DESCRIPTION
**Pull request by jsdillon**
While fixing #349, I decided to rewrite a large part of fft_dly.

1. I fixed the issue in #349 where delay units weren't being corrected
2. I used a more standard, robust, and readable approach to quadratic interpolation
3. I removed the ability to specify the time and frequency dimension in order to make the code more readable and shorter.
4. I condensed and simplified window code.
5. I made medfilt and phases default false, since they drastically slow down the code.

In the process, I noticed that the unit tests really don't cover much of the functionality of fft_dly... only that it actually produces an output. Namely, there's no test that the phase offsets are right. There's no test that the windowing or the edge_cut works. And there's no test that shows that the median filter works. I'm going to add this all as to #262.

### CODE REVIEW

**Comment by nkern** [comment]
@jsdillon thanks for these improvements and bug fixes! One thing you can do to increase readability is to get rid of all of the `np.moveaxis(np.median` calls in favor of `np.median(... keepdims=True)`. I wrote that stuff before I knew you could do `keepdims=True` :)

File: hera_cal/abscal_funcs.py [hunk 1]
```diff
@@ -996,111 +986,73 @@ def array_axis_to_data_key(data, array_index, array_keys, key_index=-1, copy_dic
...
-
-    peak_shifts = np.array([get_peak(np.take(amp, i, axis=time_ax), np.take(argmax, i, axis=time_ax)[0]) for i in range(Ntimes)])
-    dlys += np.moveaxis(peak_shifts.reshape(-1, 1) * dfreq, 0, time_ax)
+    # fft w/ window and find argmax
+    vfft = np.fft.fft(data * win * wgts, axis=1)
+    amp = np.abs(np.fft.fftshift(vfft))
+    argmaxes = np.argmax(amp, axis=1)
+    fftfreqs = np.fft.fftshift(np.fft.fftfreq(Nfreqs, df))
+    dlys, peak_shifts = np.zeros((Ntimes, 1)), np.zeros((Ntimes, 1))
+    # use parabolic peak interpolation: https://ccrma.stanford.edu/~jos/sasp/Quadratic_Interpolation_Spectral_Peaks.html
```
Comment (nkern) [inline hera_cal/abscal_funcs.py]: A few comments on the new peak finder / shifter:
1) Is it faster or slower than the one before? The reason I adopted the previous approach was that it was about as fast as way as I could think of doing this, and its an analytic result. 2) It seems like what you've implemented is actually quite similar (`0.5 * something / something`), and 3) does it give the same or nearly similar result as the old function? I'm happy to go with this new implementation if the speed isn't significantly slower and its getting similar if not the same results as before

**Comment by nkern** [comment]
you mentioned you fixed a bug on units, can you add a test to ensure that bug is fixed?

File: hera_cal/abscal_funcs.py (hunk 1 above)
Comment (jsdillon) [reply hera_cal/abscal_funcs.py]: Timing on a (4354, 1024) complex array:

New version takes 365 ms, old version takes 332 ms. The results are identical to numerical precision, so I think we're just writing the same algorithm differently.

And for the record, the old version using the default kwargs takes 2m 57s (the 332 ms is without the medfilt or the phase finding).

**Comment by jsdillon** [comment]
Regarding the `np.moveaxis` stuff, I wasn't planning on changing the phase-finding code. If you want to simplify it when you address #262, that's cool.

Regarding the bug fix on units, by setting `solve_phase=False` in this line:

`dly, offset = hc.abscal.fft_dly(r, df, medfilt=True, kernel=(1, 5), solve_phase=False)`

I exposed the error. The problem was effectively an indentation problem: the `dlys /= df` should have been run regardless of whether solve_phase was true or not.

**Pull request merged**